1 . take a data set from attached URL https://archive.ics.uci.edu/ml/datasets/Car+Evaluation
2 . create a database called as cardataset
3 . create a table called as car with a column name given in dataset description
4 . Dump all the data into car table 
5 . try to check weather all the data is aviable inside your table or not 
6 . try to group all the data with COL1 AND count occurences of each and every record based on col1 value 
7 . Try to filter a record where col 3 value will be 4 .
8.  Try to update a col 3 value with 8 whereever you have value equal to 2 
9 . try to delete table 
10 .   Try to delete database

In [2]:
import mysql.connector as con
import os


In [3]:
db_client = con.Connect(host='localhost',user='root',passwd = 'mysqlpass',use_pure = True)


In [4]:
cursor = db_client.cursor()

In [4]:
cursor.execute("show databases")
cursor.fetchall()

[('glass',),
 ('information_schema',),
 ('mysql',),
 ('performance_schema',),
 ('sakila',),
 ('student',),
 ('sys',),
 ('world',)]

In [7]:
#2 create a database called as cardataset
table_name = 'car_data'
database_name = 'sqltask'
cursor.execute("create database {}".format(database_name))
cursor.execute("show databases")
cursor.fetchall()

[('glass',),
 ('information_schema',),
 ('mysql',),
 ('performance_schema',),
 ('sakila',),
 ('sqltask',),
 ('student',),
 ('sys',),
 ('world',)]

In [13]:
#3 create a table called as car with a column name given in dataset description
cursor.execute("create table sqltask.car_data(buying VARCHAR(30), maint VARCHAR(30),doors VARCHAR(30),persons VARCHAR(30),lug_boot VARCHAR(30), safety VARCHAR(30),class VARCHAR(20))")
cursor.execute("use sqltask")
cursor.execute("show tables")
cursor.fetchall()

[('car_data',)]

In [39]:
#4 Dump all the data into car table
import csv
with open('car_data.txt','r') as f:
    rows = csv.reader(f,delimiter='\n')
    for row in rows:
        data = row[0].split(',')
        query = "insert into sqltask.car_data values(%s,%s,%s,%s,%s,%s,%s)"
        cursor.execute(query,data)
db_client.commit()
        

In [40]:
cursor.execute("select* from car_data")
cursor.fetchall()

[('vhigh', 'vhigh', '2', '2', 'small', 'low', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'small', 'med', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'small', 'high', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'med', 'low', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'med', 'med', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'med', 'high', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'big', 'low', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'big', 'med', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'big', 'high', 'unacc'),
 ('vhigh', 'vhigh', '2', '4', 'small', 'low', 'unacc'),
 ('vhigh', 'vhigh', '2', '4', 'small', 'med', 'unacc'),
 ('vhigh', 'vhigh', '2', '4', 'small', 'high', 'unacc'),
 ('vhigh', 'vhigh', '2', '4', 'med', 'low', 'unacc'),
 ('vhigh', 'vhigh', '2', '4', 'med', 'med', 'unacc'),
 ('vhigh', 'vhigh', '2', '4', 'med', 'high', 'unacc'),
 ('vhigh', 'vhigh', '2', '4', 'big', 'low', 'unacc'),
 ('vhigh', 'vhigh', '2', '4', 'big', 'med', 'unacc'),
 ('vhigh', 'vhigh', '2', '4', 'big', 'high', 'unacc'),
 ('vhigh',

In [55]:
#5 try to check weather all the data is aviable inside your table or not
with open('car_data.txt','r') as f:
    rows = csv.reader(f,delimiter='\n')
    count = 0
    for row in rows:
        count +=1
print(count)

1728


 Executes query ---> select count(*) from car_data;  in sql workbench gave result to 1728. Hence all the records are there in the database

In [46]:
#6 try to group all the data with COL1 AND count occurences of each and every record based on col1 value 
cursor.execute("select buying,count(buying) from car_data group by buying")
cursor.fetchall()

[('vhigh', 432), ('high', 432), ('med', 432), ('low', 432)]

In [50]:
#7 Try to filter a record where col 3 value will be 4 .
cursor.execute("select * from car_data where doors=4 ")
cursor.fetchall()

[('vhigh', 'vhigh', '4', '2', 'small', 'low', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'small', 'med', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'small', 'high', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'med', 'low', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'med', 'med', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'med', 'high', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'big', 'low', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'big', 'med', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'big', 'high', 'unacc'),
 ('vhigh', 'vhigh', '4', '4', 'small', 'low', 'unacc'),
 ('vhigh', 'vhigh', '4', '4', 'small', 'med', 'unacc'),
 ('vhigh', 'vhigh', '4', '4', 'small', 'high', 'unacc'),
 ('vhigh', 'vhigh', '4', '4', 'med', 'low', 'unacc'),
 ('vhigh', 'vhigh', '4', '4', 'med', 'med', 'unacc'),
 ('vhigh', 'vhigh', '4', '4', 'med', 'high', 'unacc'),
 ('vhigh', 'vhigh', '4', '4', 'big', 'low', 'unacc'),
 ('vhigh', 'vhigh', '4', '4', 'big', 'med', 'unacc'),
 ('vhigh', 'vhigh', '4', '4', 'big', 'high', 'unacc'),
 ('vhigh',

In [54]:
#8 Try to update a col 3 value with 8 whereever you have value equal to 2 
cursor.execute("update car_data set doors='8' where doors='2'")
db_client.commit()
cursor.execute("select * from car_data where doors='8'")
cursor.fetchall()

[('vhigh', 'vhigh', '8', '2', 'small', 'low', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'small', 'med', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'small', 'high', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'med', 'low', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'med', 'med', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'med', 'high', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'big', 'low', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'big', 'med', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'big', 'high', 'unacc'),
 ('vhigh', 'vhigh', '8', '4', 'small', 'low', 'unacc'),
 ('vhigh', 'vhigh', '8', '4', 'small', 'med', 'unacc'),
 ('vhigh', 'vhigh', '8', '4', 'small', 'high', 'unacc'),
 ('vhigh', 'vhigh', '8', '4', 'med', 'low', 'unacc'),
 ('vhigh', 'vhigh', '8', '4', 'med', 'med', 'unacc'),
 ('vhigh', 'vhigh', '8', '4', 'med', 'high', 'unacc'),
 ('vhigh', 'vhigh', '8', '4', 'big', 'low', 'unacc'),
 ('vhigh', 'vhigh', '8', '4', 'big', 'med', 'unacc'),
 ('vhigh', 'vhigh', '8', '4', 'big', 'high', 'unacc'),
 ('vhigh',

In [ ]:
#9 Drop table
cursor.execute("drop table sqltask.car_data")
db_client.commit()



In [7]:
#delete database
cursor.execute("drop database sqltask")
db_client.commit()

In [8]:
cursor.execute("show databases")
cursor.fetchall()

[('glass',),
 ('information_schema',),
 ('mysql',),
 ('performance_schema',),
 ('sakila',),
 ('student',),
 ('sys',),
 ('world',)]

The database "sqltask" is deleted as shown above